nodejs/week3/meal_sharing_app/src/backend/api/meals.js

```diff
@@ -0,0 +1,142 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+const knex = require("../database");
+
+app.use(express.json());
+app.use(express.urlencoded({
+  extended: true
+}));
+
+//GET api/meals/ query parameters
+
+router.get("/", async function (req, res) {
+  const {
+    maxPrice,
+    title,
+    availableReservations,
+    createdAfter,
+    limit
+  } = req.query;
+  const mealTitles = await knex.select("*").table("meal");
+
+  if (Object.keys(req.query).length === 0) {
+    res.send(mealTitles);
+
+  }
+
+  if (maxPrice) {
+    try {
+
+      const maxPriceMeals = await knex.select("*").table("meal")
+        .where("price", "<", maxPrice);
+      console.log(maxPriceMeals);
+      res.send(maxPriceMeals);
+    } catch (error) {
+      error;
+    }
+  } else if (title) {
+    try {
+
+      const titles = await knex.select("*").table("meal")
```

Comment:
Again you are using titles, but the are meals. I would maybe call them queriedMeals. So we indicate that these meals have been queried (searched for)